tsconfig.json

```diff
@@ -19,8 +19,14 @@
 		"noUnusedParameters": true,            /* Report errors on unused parameters. */
 		"noImplicitReturns": true,             /* Report error when not all code paths in function return a value. */
 		"noFallthroughCasesInSwitch": true,    /* Report errors for fallthrough cases in switch statement. */
+
+		"baseUrl": ".",
+		"paths": {
+			"@wordpress/*": [ "./packages/*/src" ]
+		}
```

Comment:
This fixes the issues with the unfound `@wordpress/dom-ready` module and doesn't seem to introduce any issues.

Seems sufficient for now.

Reply:
> This fixes the issues with the unfound `@wordpress/dom-ready` module and doesn't seem to introduce any issues.
> 
> Seems sufficient for now.

Related: #18927

Once this is in place, we'll probably have trouble adding types to anything which depends on another package which isn't type-ready (i.e. if there are any errors in the dependency code).

Reply:
> we'll probably have trouble adding types to anything which depends on another package which isn't type-ready 

Based on my experience here, it's already a problem because tsc can't locate the modules.

Reply:
Would you prefer to sort that out in #18927 or shall we include it here?

Reply:
> Would you prefer to sort that out in #18927 or shall we include it here and merge this?

I've merged #18927 if you want to go ahead and rebase it here.